### DISCUSSION TITLE
Customtkinter - 'back' button not working

### DISCUSSION THREAD

**Discussion by ShowzZzie** [TomSchimansky/CustomTkinter#988] [Q&A]
Hey,

First time creating a project with GUI, I ran into a problem, where back_button in notes_page returns only part of main(). On startup, it works correctly, there's a clock in the center a little message and two buttons, but when i click back_button inside of notes_page the home page shows only clock and message but no buttons, nor their text, nor their images.

`customtkinter.set_appearance_mode("dark")
customtkinter.set_default_color_theme("dark-blue")

# Create the main window
app = customtkinter.CTk()
app.geometry('1000x650')
app.resizable(False, False)

def update_time():
    current_time = datetime.now()
    time_str = current_time.strftime('%I:%M:%S %p')

    app.after(1000, update_time)  # Schedule the function to be called after 1 second
    # add time to the top of the app
    current_time = datetime.now()
    time = current_time.strftime('%I:%M:%S %p')
    date = customtkinter.CTkLabel(master=app, text=time, font=('small fonts', 40, 'bold'))
    date.configure(text=time_str)
    date.place(x=510, y=50, anchor='c')

# Create lists to store notes and reminders
notes = []
reminders = []

def remove_widgets(widget):
    for child in widget.winfo_children():
        child.destroy()

def home():

    update_time()

    # Add some text to greet the user
    customtkinter.CTkLabel(master=app, text='Welcome back!\nUse buttons below to navigate', font=('helvetica', 30, 'bold')).place(x=300, y=125)

    # Load the images for the buttons
    image1 = Image.open('button1.jpg')
    button1_image = customtkinter.CTkImage(image1, size=(150, 150))

    image2 = Image.open('button2.jpg')
    button2_image = customtkinter.CTkImage(image2, size=(150, 150))

    # Create the buttons with images and text
    button1 = customtkinter.CTkButton(master=app, text='\nNotes', font=('Helvetica', 24, 'bold'), image=button1_image, width=300, height=300, compound='top', command=notes_page)
    button2 = customtkinter.CTkButton(master=app, text='\nReminders', font=('Helvetica', 24, 'bold'), image=button2_image, width=300, height=300, compound='top', command=reminders_page)

    # Configure the rows and columns to have equal weight
    app.columnconfigure(0, weight=1)
    app.columnconfigure(1, weight=1)
    app.rowconfigure(0, weight=1)

    width = app.winfo_reqwidth()
    height = app.winfo_reqheight()
    new_w = width//2
    new_h = height//2

    # Place the buttons in the first two columns of the first row
    button1.place(x=new_w+55, y=new_h+150)
    button2.place(x=new_w*5+55, y=new_h+150)

    update_time()

# Define functions for the buttons
def notes_page():

    remove_widgets(app)

    def delete_note(note):
        notes.remove(note)
        notes_page()

    # Create a frame to hold the notes
    notes_frame = tk.Frame(app, width=1250, height=9999, bg='#1a1a1a')
    notes_frame.pack()

    # Create a canvas to hold the notes
    canvas = tk.Canvas(notes_frame, width=1250, height=9999, bg='#1a1a1a', highlightthickness=0)
    canvas.pack(side='left', fill='both', expand=True)

    # Create a scrollbar for the canvas
    scrollbar = tk.Scrollbar(notes_frame, orient='vertical', command=canvas.yview)
    scrollbar.place(x=1230, y=0, height=750)

    # Set the scrollbar as the canvas's yview
    canvas.configure(yscrollcommand=scrollbar.set)

    # Create a frame to hold the widgets for the notes
    widgets_frame = tk.Frame(canvas, width=1250, height=9999, bg='#1a1a1a')

    # Add the widgets frame to the canvas
    canvas.create_window((0, 0), window=widgets_frame, anchor='nw')

    # Bind the scroll event to the canvas
    widgets_frame.bind('<Configure>', lambda event, canvas=canvas: on_configure(event, canvas))

    back_button = customtkinter.CTkButton(master=canvas, text='Back', font=('helvetica', 16, 'bold'), width=150, height=50,
                                          command=home)
    back_button.place(x=0, y=0)

    new_note_button = customtkinter.CTkButton(master=canvas, text='Add', font=('helvetica', 16, 'bold'), width=150,
                                              height=50, command=add_note, fg_color='#1AA600', hover_color='#168D00')
    new_note_button.place(x=850, y=600)

    height = 0

    for i, note in enumerate(sorted(notes, key=lambda x: x['priority'], reverse=True)):
        if note['priority'] == 3:
            imageIM = Image.open('3.png')
            image = customtkinter.CTkImage(imageIM, size=(30, 30))
        elif note['priority'] == 2:
            imageIM = Image.open('2.png')
            image = customtkinter.CTkImage(imageIM, size=(30, 30))
        else:
            imageIM = Image.open('1.png')
            image = customtkinter.CTkImage(imageIM, size=(30, 30))

        label = customtkinter.CTkLabel(master=widgets_frame, image=image, text=f" {note['title']}", font=('Helvetica', 24, 'bold'), compound='left', padx=5)
        label.place(x=0, y=100 + height)
        delete_button = customtkinter.CTkButton(master=widgets_frame, text='Delete', font=('Helvetica', 16, 'bold'), width=150, height=30, command=lambda n=note: delete_note(n), fg_color='#C30000', hover_color='#A30000')
        delete_button.place(x=820, y=100+height)
        label = customtkinter.CTkLabel(master=widgets_frame, text=note['desc'], font=('Helvetica', 18), wraplength=999, justify='left', padx=5)
        label.place(x=0, y=135 + height)
        length = len(note['desc'])
        num_lines = math.ceil(length / 320)
        gap_size = 70  # fixed gap size
        height += num_lines * 35
        if i < len(notes) - 1:  # add gap only if not the last note
            height += gap_size
        else:
            height += gap_size + 25

    canvas.configure(scrollregion=canvas.bbox('all'))

def on_configure(event, canvas):
    """Reset the canvas's scroll region to encompass the inner frame"""
    canvas.configure(scrollregion=canvas.bbox('all'))`
    
    
I also have a reminders_page in which button is implemented in the exact same way and clicking back works perfectly there, grateful for any hints, tips etc.

**Comment by abhaymakes** [comment]
Hey @ShowzZzie, I'm not sure about customtkinter(it still has several bugs i guess) but, you could achieve the same thing in tkinter and maybe make some tweaks to make it look like customtkinter, [this blog](https://www.google.com/amp/s/blog.teclado.com/tkinter-scrollable-frames/amp/) might help you or [this video](https://youtu.be/0WafQCaok6g) whichever you prefer. Ask me if this doesn't solve your issue.

**Reply by ShowzZzie** [reply]
Hey, i managed to get scrollbar working, thanks for help nontheless, appreaciate it! <3
Right now however it presented different bug ;p
